libs/src/services/creatorNode/CreatorNodeSelection.js

```diff
@@ -0,0 +1,249 @@
+   */
+  async select () {
+    // Reset decision tree and backups
+    this.decisionTree = []
+    this.clearBackups()
+
+    // Get all the creator node endpoints on chain and filter
+    let services = await this.getServices()
+    this.decisionTree.push({ stage: 'Get All Services', val: services })
+
+    if (this.whitelist && this.whitelist.length > 0) { services = this.filterToWhitelist(services) }
+    this.decisionTree.push({ stage: 'Filtered To Whitelist', val: services })
+
+    if (this.blacklist && this.blacklist.length > 0) { services = this.filterFromBlacklist(services) }
+    this.decisionTree.push({ stage: 'Filtered Out From Blacklist', val: services })
+
+    const healthCheckedServices = await this.timeRequests(
+      services.map(node => ({
+        id: node,
+        url: `${node}/health_check`
+      }))
+    )
+    // Store a copy of the sorted by version and resposne time nodes
+    this.healthCheckedServicesList = healthCheckedServices
+
+    const healthyServices = healthCheckedServices.filter(resp => {
+      const endpoint = resp.request.id
+
+      // Add everything as backup regardless of response
+      this.addBackup(endpoint, resp.response)
+
+      let isHealthy = false
+      if (resp.response) {
+        const isUp = resp.response.status === 200
+        const versionIsUpToDate = this.ethContracts.hasSameMajorAndMinorVersion(this.currentVersion, resp.response.data.data.version)
+        isHealthy = isUp && versionIsUpToDate
+      }
+
+      // If not healthy, remove as backup and add to unhealthy
+      if (!isHealthy) {
+        this.removeFromBackups(endpoint)
+        this.addUnhealthy(endpoint)
+      }
+      return isHealthy
+    })
+    services = healthyServices.map(service => service.request.id)
+
+    this.decisionTree.push({ stage: 'Filtered Out Known Unhealthy And Outdated Nodes', val: services })
+
+    const successfulSyncCheckServices = []
+    for (const service of services) {
+      let status
+      try {
+        status = await this.creatorNode.getSyncStatus(service)
+      } catch (e) {
+        // Error with checking sync status -- deem current service as unhealthy
+        console.warn(`CreatorNodeSelection - Err with checking sync status for ${service}: ${e}`)
+        this.removeFromBackups(service)
+        this.addUnhealthy(service)
+        continue
+      }
+
+      const { isBehind, isConfigured } = status
+
+      // a first time creator will have a sync status as isBehind = true and isConfiugred = false. this is ok
+      const firstTimeCreator = isBehind && !isConfigured
+      // an existing creator will have a sync status (assuming healthy) as isBehind = false and isConfigured = true. this is also ok
+      const existingCreator = !isBehind && isConfigured
+      // if neither of these two are true, the cnode is not suited to be selected
+      if (firstTimeCreator || existingCreator) successfulSyncCheckServices.push(service)
```

Comment:
tested this by checking what the console spit out on staging when:
1. am not a creator
2. am a creator
```
const { isBehind, isConfigured } = await audiusLibs.creatorNode.getSyncStatus(node.endpoint)
```
and the values from the console are reflected in the comments

Reply:
but not too sure if this is how we want to approach this

Reply:
@raymondjacobson do we want to go with this approach? seems kinda brittle